sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonParser.scala

```diff
@@ -89,6 +95,24 @@ class JacksonParser(
 
   /**
    * Create a converter which converts the JSON documents held by the `JsonParser`
+   * to a value according to a desired schema. This is an overloaded method to the
+   * previous one which allows to handle array of primitive types.
+   */
+  private def makeRootConverter(at: ArrayType): JsonParser => Seq[Any] = {
+    (parser: JsonParser) => parseJsonToken[Seq[Any]](parser, at) {
+      case START_ARRAY =>
+        val array = convertArray(parser, makeConverter(at.elementType))
+        if (array.numElements() == 0) {
+          Nil
+        } else {
+          array.toArray(at.elementType).toSeq
+        }
+      case _ => Nil
```

Comment:
Should we return `Nil` when it is not parsed to array? The original `makeRootConverter` didn't do this.

Reply:
You are right, I will remove this.